### PR TITLE
Diana development

### PR DESCRIPTION
**Pull request by dianamstan**

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -38,7 +38,7 @@
     "@types/webpack": "4.4.31",
     "@types/webpack-dev-middleware": "2.0.2",
     "apollo-link-schema": "1.2.2",
-    "axios": "0.18.0",
```
Comment (dianamstan) [inline package.json]: not sure how these changes happened... or what `package-lock.json` is above

File: server/models/item.ts [hunk 2]
```diff
@@ -0,0 +1,113 @@
+const { Model } = require('objection');
+import { Transaction } from 'objection';
+import { Pokemon } from './pokemon';
+
+interface IItemCreateInput {
```
Comment (dianamstan) [inline server/models/item.ts]: mistakenly didn't start a new branch so my old commit is included in these changes? will make a new branch next time

File: server/models/migrations/20191001141358_initial-pokedex-migration.js [hunk 3]
```diff
@@ -0,0 +1,52 @@
+exports.up = function(knex, Promise) {
```
Comment (dianamstan) [inline server/models/migrations/20191001141358_initial-pokedex-migration.js]: not sure how to get rid of this old commit from looking new still :(

File: server/models/item.ts [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+  }
+
+  static async getAll(txn: Transaction): Promise<Item[]> {
+    return this.query(txn);
+  }
+
+  static async create(input: IItemCreateInput, txn: Transaction): Promise<Item> {
+    const createdItem = this.query(txn).insertAndFetch(input);
+    if (!createdItem) {
+      return Promise.reject(`Could not create Item, please try again.`);
```
Comment (raletty) [inline server/models/item.ts]: Might be useful to log some info on what was attempted in creation.

File: server/models/item.ts [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+    }
+    return createdItem;
+  }
+
+  static async edit(itemId: string, item: IItemEditInput, txn: Transaction): Promise<Item> {
+    const editedItem = this.query(txn).patchAndFetchById(itemId, item);
+    if (!editedItem) {
+      return Promise.reject(`No such item: ${itemId}`);
```
Comment (raletty) [inline server/models/item.ts]: Would add something distinguishing in these messages (i.e. for `get()`, `delete()` etc).

File: server/models/item.ts (hunk 4 above)
Comment (dianamstan) [reply server/models/item.ts]: ah, i had logging info locally but didn't commit it. in general, are we ok to commit logging in tests ?

**Comment by dianamstan** [comment]
my tests are failing in production but not locally - i think the issue is the test database being used by the circleci tests is different than the version i had locally. so assumign taht what ill have to do is create a new pokemon instance for each test because i can't depend on the state of the circleci database

**Comment by kirilllevin** [comment]
I think you can ignore the CircleCI failures. As you suggest, it's related to inconsistent data. The main issue is that we mint uuids on insert, which means that each environment will have its own ids, but the tests require ids to be hardcoded. A solution to this would be to use consistent seed data (including ids) in all the setup, but that might be somewhat involved, and it's not clear whether it's worth the effort for anybody to fix.

**Comment by dianamstan** [comment]
oh nice, looks like i can merge it into my `diana-pokedex` branch even with the test failures

**Pull request merged**